syft/internal/cache/error_resolver.go

```diff
@@ -0,0 +1,40 @@
+package cache
+
+import "fmt"
+
+// GetResolverCachingErrors returns a Resolver that caches errors and will return them
+// instead of continuing to call the provided resolve functions
+func GetResolverCachingErrors[T any](name string) Resolver[T] {
```

Comment:
It might be better to say: this should be handled by the resolver function, but then I foresee all the implementations duplicating something more-or-less like this, so this is just here for a convenience